### PR TITLE
Use systemd from udev and instead of pmutils

### LINKED ISSUE #61: udev rule executed too early?
After getting the udev rule to work, I am *sometimes* running into trouble, where disconnecting an external monitor changes the profile to external monitor, and connecting an external monitor changes the profile to only my laptop. So it's doing the *opposite* of what it should).

My guess is that `autorandr` is run too "early", such that the output of `xrandr` does not reflect the actual state of things.

If I have time one of the next days, I'll try to look into it.

### PR DESCRIPTION
**Pull request by phillipberndt**
This is an idea for how we could resolve #61 and get rid of the pmutils scripts on Debian.

What would change by this PR:

* pmutils script does no longer get installed by default if systemd is in use; the systemd unit already executes autorandr upon waking from sleep
* the udev rule now uses systemd to start autorandr, allowing for longer runtimes without triggering any kill timeouts (udev kills long running processes)

Auxiliary changes:

* the Makefile now adjusts paths in the scripts/config files depending on the `$PREFIX`
* the udev rule is now dynamically created, calling either systemd or autorandr directly if systemd is unavailable

This affects #60 as well.

### CODE REVIEW

**Comment by RasmusWL** [comment]
Thanks for looking into this! I'll try to do some testing during this week! (still dealing with jetlag)

**Comment by phillipberndt** [comment]
Any objections against merging this? (cc @blueyed)

**Comment by blueyed** [comment]
I've done no testing though, so maybe wait for @RasmusWL ?!

**Comment by RasmusWL** [comment]
I'm back in business, and I'll be able to test this one of the coming days!

File: contrib/pm-utils/40autorandr [hunk 1]
```diff
@@ -5,6 +5,6 @@ exec > /var/log/autorandr.log 2>&1
 
 case "$1" in
 	thaw|resume)
-		autorandr --batch -c --default default
+		/usr/bin/autorandr --batch --change --default default
```
Comment (Flowdalic) [inline contrib/pm-utils/40autorandr]: Why the switch to an absolute path to the `autorandr` binary?

Comment (phillipberndt) [inline contrib/pm-utils/40autorandr]: The change is accompanied by a change in the Makefile,

    sed -i -re 's#/usr/bin/autorandr#$(subst #,\#,${PREFIX})/bin/autorandr#g' ${DESTDIR}/${PM_SLEEPHOOKS_DIR}/40autorandr

The general idea is that this allows users to install autorandr to a path outside of the system's $PATH.

**Comment by phillipberndt** [comment]
Any objections, @RasmusWL?

**Comment by RasmusWL** [comment]
@phillipberndt sorry I have been so late responding to this. I know I promised several times, and I have been thinking about this a lot since then!

Anyway, without futher ado, here is my feedback:

This sadly doesn't solve #61 :cry: . But it is a bit different from what I experienced when I submitted that issue (back then it would switch between the two profiles, but apply them at the wrong time).

**TL;DR;** on a clean reboot, it worked the first time I disconnected and reconnected, from there on only my laptop screen is turned on, with occasional all monitors off.

I started of with a profile loaded that only have my external monitor on (over DisplayPort). From here on is what happened when I disconnected and reconnected the plug:
* disconnecting works good (back to laptop)
* reconnecting worked good (external monitor lid up)
* Then disconnecting again was VERY BAD (all monitors off)
* reconnecting was bad (turned on laptop)
* disconnecting was good (left laptop on)
* reconneting was bad (left laptop on)
* disconnecting was VERY BAD (turned all off)
* reconnecting was bad (turned on laptop)

At this point I ran autorandr manually, which changed the setup to the correct one, only using the external monitor.
* disconn = laptop (good)
* reconn = laptop (bad)
* disconn = laptop (good)
* reconn = latop (bad)
* disconn = all black (very-bad)
* reconn = laptop (good)

```
$ xrandr --version 
xrandr program version       1.5.0
Server reports RandR version 1.5
```

**Comment by RasmusWL** [comment]
I just tried to use the most recent version of `master`, and I actually get the same results as described above. When disconnecting and reconnecting my plug, I either get the laptop monitor on (most of the time), or both monitors are turned off.

**Comment by phillipberndt** [comment]
Did you try with a `predetect` script containing e.g. `sleep 5` or without?

**Comment by RasmusWL** [comment]
No I did not. I forgot it existed. I can try it just now!

One wierd thing. I think the `postswitch` hook in my 'tv' profile was run (probably during my testing), because I just noticed my soundcard was set to play over HDMI... and it definitely was not set to that before.

**Comment by RasmusWL** [comment]
the `predetect` script is run *every* time, also when running `autorandr`. I suppose this is not intended.

I also (sadly) have to report that adding the predetect script did not work (I had `sleep 5` in it)

**Comment by phillipberndt** [comment]
That is intended - there were users in the past (back with werarbytes repo) that required to run `xrandr -q` twice every time. But you can detect the batch run using ps/pgrep/grep. Here's the version I'm using, maybe this one (or a variant) will work:

```sh
#!/bin/sh
if pgrep -af autorandr | grep -q batch; then
	sleep 2
	xrandr -q >/dev/null 2>&1
	sleep .5
	xrandr -q >/dev/null 2>&1
	sleep .5
fi
```

**Comment by blueyed** [comment]
> That is intended - there were users in the past (back with werarbytes repo) that required to run xrandr -q twice every time.

But not for `--help` though.. :)

**Comment by phillipberndt** [comment]
Good point. Fixed in the predetect branch, it processes `--help` first now. I've also added an environment variable to make it easier to detect batch invocation.

All right.. so I understand that even though these changes do not fix @RasmusWL's issue, they don't do any harm either, right? They get rid of the pmutils dependency on modern systems, which I consider an improvement worth having; I'd hence still like to merge this.

**Comment by blueyed** [comment]
@phillipberndt 
I've not reviewed it since my last LGTM, but what you say makes sense.. :)

**Comment by RasmusWL** [comment]
Great stuff, thanks for looking into it.

I also wanted to mention that I think this it is an acceptable fix to #60.

ps. changelog dates are listed in the wrong order now :)
